### PR TITLE
Fix for Issue-421

### PR DESCRIPTION
**Pull request by danielsn**
*Issue #, if available:*
https://github.com/aws/aws-encryption-sdk-c/issues/421

*Description of changes:*
Fix the issue https://github.com/aws/aws-encryption-sdk-c/issues/421

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/enc_ctx.c [hunk 1]
```diff
@@ -64,29 +65,25 @@ int aws_cryptosdk_enc_ctx_size(size_t *size, const struct aws_hash_table *enc_ct
...
+    if (!aws_byte_buf_write_be16(output, (uint16_t)num_elems)) {
+        aws_byte_buf_clean_up(output);
         return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
     }
 
-    if (length == 0) {
-        // Empty encryption context
-        return AWS_OP_SUCCESS;
-    }
+    if (num_elems == 0) return AWS_OP_SUCCESS;
```
Comment (ttjsu-aws) [inline source/enc_ctx.c]: If the num_elems is zero we want to add a value 0 to the header indicating that there are no elements. Line 71-72 is initializing  `output` to 0. This needs to be added only if the num_elems is 0 though.

Comment (danielsn) [reply source/enc_ctx.c]: Line 71-72 is writing whatever the ```num_elems``` happens to be, whether that is ```0``` or non-zero.  If its zero, we've written our 0 and are done; if it is non-zero, we also, need to output the pairs themselves, which is what the loop below does.

Comment (ttjsu-aws) [reply source/enc_ctx.c]: sgtm. 
Could you run the format script on this? 
Our styling for the if statement is different from this one.

File: source/enc_ctx.c [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include <aws/cryptosdk/private/enc_ctx.h>
 #include <aws/cryptosdk/private/utils.h>
 
+#include <aws/common/assert.h>
```
Comment (dougch) [inline source/enc_ctx.c]: Tests are failng to find this file, origin?

**Comment by danielsn** [comment]
#422 superedes this

**Pull request closed without merging**